ray-operator/controllers/ray/utils/ray_resources.go

```diff
@@ -0,0 +1,137 @@
+package utils
+
+import (
+	"strconv"
+	"strings"
+
+	rayiov1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+
+	v1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var rayClusterLog = logf.Log.WithName("RayCluster-Controller")
+
+// Updates the user-specified rayResource spec with CPU, GPU, and memory data from the rayStartParams and the pod template.
+// CPU, GPU, and memory data from user-specified rayResourceSpec overrides data from rayStartParams.
+// CPU, GPU, and memory data from rayStartParams overrides data from the podTemplate.
+// If CPU, GPU, or memory is not present in the resource map returned by this method, Ray will attempt to
+// infer these upon "ray start".
+// The map returned by this method is used in two ways:
+// 1. It is passed as an environment variable to the Ray node. This advertises resource capacities to the Ray scheduler.
+// 2. It is written to the RayCluster CR's Status field. This allows the Ray autoscaler to access the resource capacities.
+func ComputeRayResources(
```

Comment:
Seems like there's a log of duplicate logic in the default computation functions, can we condense them into a single `func k8s_resource_allocation_to_ray_resources ( rayContainerResources v1.ResourceRequirements, resource_type string) (int64)`